src/utils/automation-helper.js

```diff
@@ -32,6 +32,9 @@ export const getAutomation = async (automationId) => {
 
 export const getAutomationRun = async (automationId, runId) => {
   const response = await fetch(`../api/v1/automations/${automationId}/runs/${runId}`);
+  if (response.status !== 200) {
```

Comment:
What kind of response code (status) is returned when the user does not have permissions? If 403, you should consider to only check for that